### ISSUE TITLE
panic when releases.files is NULL

### ISSUE THREAD

**Issue by jyn514** [rust-lang/docs.rs#517]
Caught this from looking through the logs on the build server: https://docs.rs/crate/rips/0.2.0/source/ gives a 500 when you visit it. The build for `rips` failed, maybe that's why the files are NULL?

Backtrace:

```
web_1  | thread '<unnamed>' panicked at 'error retrieving column 5: Error(Conversion(WasNull))', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/postgres-0.15.2/src/rows.rs:201:31
web_1  | stack backtrace:
web_1  |   13: postgres::rows::Row::get
web_1  |   14: cratesfyi::web::source::source_browser_handler
web_1  |   15: <F as iron::middleware::Handler>::handle
web_1  |   16: <alloc::boxed::Box<dyn iron::middleware::Handler> as iron::middleware::Handler>::handle
web_1  |   17: router::router::Router::handle_method
web_1  |   18: <router::router::Router as iron::middleware::Handler>::handle
web_1  |   19: iron::middleware::Chain::continue_from_handler
web_1  |   20: <iron::middleware::Chain as iron::middleware::Handler>::handle
web_1  |   21: <alloc::boxed::Box<dyn iron::middleware::Handler> as iron::middleware::Handler>::handle
web_1  |   22: <cratesfyi::web::CratesfyiHandler as iron::middleware::Handler>::handle
web_1  |   23: <iron::iron::RawHandler<H> as hyper::server::Handler>::handle
web_1  |   24: hyper::server::Worker<H>::handle_connection
web_1  |   25: hyper::server::listener::spawn_with::{{closure}}
```

Note for maintainers: to find other crates with this issue, run this psql command on the server:

`SELECT releases.target_name, releases.version FROM releases WHERE releases.files IS NULL;`

**Comment by jyn514** [comment]
It looks like this happens whenever the build fails.

**Comment by jyn514** [comment]
Relevant code: https://github.com/rust-lang/docs.rs/blob/master/src/web/source.rs#L107

**Comment by jyn514** [comment]
Hmm ... I worked around this by returning `None` in the linked code, which avoids the panic, but the rendered page is still mostly blank :(

![mostly-blank_1920x1080](https://user-images.githubusercontent.com/23638587/70746155-6710b400-1cf3-11ea-9f0f-35c02843e8c7.png)

**Comment by jyn514** [comment]
I turned it into a 404 instead, at least we're giving _an_ error message now.

**Comment by jyn514** [comment]
Reopening because of #521

**Comment by jyn514** [comment]
While fixing this, I noticed that using the current master branch, binaries do _not_ have their source code saved into the database. However, if I go to e.g. https://docs.rs/crate/bat/0.12.1/source/ all the code is there. Maybe the builder got changed at some point? If so I think it makes more sense to store the code in the DB even for binaries than to special case the error handling if it's not there (see also https://github.com/rust-lang/docs.rs/issues/520).

cc @pietroalbini , could this have happened during the switch to rustwide?

**Comment by pietroalbini** [comment]
Not sure if I see the point of storing the source code for binaries... people shouldn't look on docs.rs for them.

**Comment by jyn514** [comment]
Ok, in that case I made a PR to fix the panic.

**Comment by jyn514** [comment]
Fixed in #526
